src/main/java/edu/byu/cs/dataAccess/sql/helpers/SqlReader.java

```diff
@@ -136,14 +136,15 @@ private void setValue(PreparedStatement ps, int wildcardIndex, T item, ColumnDef
      */
     public void setValue(@NonNull PreparedStatement ps, int wildcardIndex, @Nullable Object value) throws SQLException {
         // This represents all the supported column types
-        if (value == null) ps.setNull(wildcardIndex, NULL);
-        else if (value instanceof String v) ps.setString(wildcardIndex, v);
-        else if (value instanceof Integer v) ps.setInt(wildcardIndex, v);
-        else if (value instanceof Float v) ps.setFloat(wildcardIndex, v);
-        else if (value instanceof Boolean v) ps.setBoolean(wildcardIndex, v);
-        else if (value instanceof Timestamp v) ps.setTimestamp(wildcardIndex, v);
-        else if (value instanceof Object v) ps.setObject(wildcardIndex, v);
-        else throw new RuntimeException("Unsupported type of value: " + value);
+        switch (value) {
```

Comment:
@pawlh I figured out that my environment didn't suggest this change because it was running Java 20. In fact, it complained about this feature because this language feature was introduced in Java 21. After I upgraded my version to Coretto 21, it worked just great.

As long as we're clear that Java 21 is the oldest version we support, then this is definitely a much cleaner syntax than the previous edition.